sdk/src/main/java/com/oursky/authgear/app2app/App2App.kt

```diff
@@ -186,33 +210,22 @@ internal class App2App(
                 // Can't parse redirect_uri, throw the error
                 throw e
             }
-            var error = "unknown_error"
-            var errorDescription: String? = (e.message ?: "Unknown error")
-            when (e) {
-                is OAuthException -> {
-                    error = e.error
-                    errorDescription = e.errorDescription
-                }
-                is ServerException -> {
-                    error = "server_error"
-                    errorDescription = e.message
-                }
-            }
-            val query = hashMapOf(
-                "error" to error
-            )
-            if (errorDescription != null) {
-                query["error_description"] = errorDescription
-            }
-            val errorURI = redirectURI.buildUpon()
-                .encodedQuery(query.toQueryParameter())
-                .build()
+            val errorURI = constructErrorUri(redirectURI, "unknown_error", e)
             val intent = Intent(Intent.ACTION_VIEW, errorURI)
             intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
             application.startActivity(intent)
         }
     }
 
+    @RequiresApi(Build.VERSION_CODES.M)
+    suspend fun rejectApp2AppAuthenticationRequest(request: App2AppAuthenticateRequest, reason: Throwable) {
+        val redirectURI = Uri.parse(request.redirectUri)
+        val errorURI = constructErrorUri(redirectURI, "invalid_grant", reason)
```

Comment:
I think the default error should not be `invalid_grant`. Maybe use `x_app2app_rejected`